.github/workflows/release.yaml

```diff
@@ -0,0 +1,131 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      binary-to-release:
+        description: Binary to release
+        type: choice
+        options:
+          - ampd
+          - agggregate-verifier
+          - batching
+          - connection-router
+          - gateway
+          - multisig
+          - multisig-prover
+          - nexus-gateway
+          - rewards
+          - service-registry
+          - voting-verifier
+      dry-run:
+        description: Dry run
+        type: boolean
+        default: true
+
+jobs:
+
+  release:
+    name: Release ${{ github.event.inputs.binary-to-release }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: '0'
+      - name: Setup variables for sub-project to release
```

Comment:
Maybe we can use lookup tables here instead of conditionals to make this more readable. 

Hashmap support is not great in bash, a workaround I can think of is use heredoc to construct the map in json string and jq to parse and access keys/values. 

`hashmap=$(cat <<EOF
{
  "ampd": {  
    "binary-to-release": "ampd",  
    "major-pattern": "MAJOR-AMPD",  
    "minor-pattern": "MINOR-AMPD",  
  }
}
EOF
)`